skimage/io/_plugins/matplotlib_plugin.py

```diff
@@ -147,9 +149,17 @@ def imshow(im, *args, **kwargs):
     kwargs.setdefault('cmap', cmap)
     kwargs.setdefault('vmin', lo)
     kwargs.setdefault('vmax', hi)
-    ax_im = plt.imshow(im, *args, **kwargs)
+    if 'ax' in kwargs:
+        ax = kwargs.pop('ax')
+    else:
+        ax = plt.gca()
+    ax_im = ax.imshow(im, *args, **kwargs)
     if cmap != _default_colormap:
-        plt.colorbar()
+        divider = make_axes_locatable(ax)
+        cax = divider.append_axes("right", size="5%", pad=0.05)
```

Comment:
Would it be possible (and at all useful) to expose `size` and `pad`, rather than hardcoding them? 

Feel free to argue against - simpler API is usually better if modifying these is a rare use case.